File: ws/paucar/u2/WS15/ZooApp/src/main/java/ec/edu/espe/zooapp/model/Animal.java
package ec.edu.espe.zooapp.model;

import java.util.ArrayList;
import java.util.Date;

/**
 *
 * @author Dennis Paucar
 */
public class Animal {
    private int id;
    private String name;
    private Date bornOnDate;
    private float weight;
    private char sex;
    private ArrayList<Vaccine> vaccines;
    
    public void feed(Food food){
        System.out.println(" Feeding the animal " + name + "with -->" + food);
    }
    
    public void register(){
        System.out.println("Registering the animal" + name);
    }
    
    public void assignCage(Cage cage){
        //TODO assign this animal to a cage
        System.out.println("Assigning an animal " + name + "to cage -->" + cage);
        cage.getAnimals().add(this);
        
    }
    
    public void provideVaccination(Vaccine vaccine){
        System.out.println(" providing vaccine -->" + vaccine + "<-- to" + name);
        vaccines.add(vaccine);
    }
    

    public Animal(int id, String name, Date bornOnDate, float weight, char sex, ArrayList<Vaccine> vaccines) {
        this.id = id;
        this.name = name;
        this.bornOnDate = bornOnDate;
        this.weight = weight;
        this.sex = sex;
        this.vaccines = vaccines;
    }

    @Override
    public String toString() {
        return "Animal{" + "id=" + id + ", name=" + name + ", bornOnDate=" + bornOnDate + ", weight=" + weight + ", sex=" + sex + ", vaccines=" + vaccines + '}';
    }

    
    
    /**
     * @return the id
     */
    public int getId() {
        return id;
    }

    /**
     * @param id the id to set
     */
    public void setId(int id) {
        this.id = id;
    }

    /**
     * @return the name
     */
    public String getName() {
        return name;
    }

    /**
     * @param name the name to set
     */
    public void setName(String name) {
        this.name = name;
    }

    /**
     * @return the bornOnDate
     */
    public Date getBornOnDate() {
        return bornOnDate;
    }

    /**
     * @param bornOnDate the bornOnDate to set
     */
    public void setBornOnDate(Date bornOnDate) {
        this.bornOnDate = bornOnDate;
    }

    /**
     * @return the weight
     */
    public float getWeight() {
        return weight;
    }

    /**
     * @param weight the weight to set
     */
    public void setWeight(float weight) {
        this.weight = weight;
    }

    /**
     * @return the sex
     */
    public char getSex() {
        return sex;
    }

    /**
     * @param sex the sex to set
     */
    public void setSex(char sex) {
        this.sex = sex;
    }

    /**
     * @return the vaccines
     */
    public ArrayList<Vaccine> getVaccines() {
        return vaccines;
    }

    /**
     * @param vaccines the vaccines to set
     */
    public void setVaccines(ArrayList<Vaccine> vaccines) {
        this.vaccines = vaccines;
    }
    
    
    
}

File: hw/pilatasig/u2/ZooApp JavaScript/ec/edu/espe/zoo/model/Animal.js
export class Animal {
    constructor(id, name, bornOnDate, weight, sex, vaccines = []) {
        this._id = id; // int
        this._name = name; // String
        this._bornOnDate = bornOnDate; // Date
        this._weight = weight; // float
        this._sex = sex; // char
        this._vaccines = vaccines; // Array of Vaccine objects
    }

    // Getters
    get id() {
        return this._id;
    }

    get name() {
        return this._name;
    }

    get bornOnDate() {
        return this._bornOnDate;
    }

    get weight() {
        return this._weight;
    }

    get sex() {
        return this._sex;
    }

    get vaccines() {
        return this._vaccines;
    }

    set id(value) {
        this._id = value;
    }

    set name(value) {
        this._name = value;
    }

    set bornOnDate(value) {
        this._bornOnDate = value;
    }

    set weight(value) {
        this._weight = value;
    }

    set sex(value) {
        this._sex = value;
    }

    set vaccines(value) {
        this._vaccines = value;
    }

    toString() {
        return `Animal{id=${this._id}, name=${this._name}, bornOnDate=${this._bornOnDate}, weight=${this._weight}, sex=${this._sex}, vaccines=${this._vaccines.map(v => v.toString()).join(", ")}}`;
    }

    feed(food) {
        console.log(`Feeding the animal ${this._name} with --> ${food}`);
    }

    register() {
        console.log(`Registering the animal ${this._name}`);
    }

    assignCage(cage) {
        console.log(`Assigning an animal ${this._name} to cage --> ${cage}`);
        cage.animals.push(this);
    }

    provideVaccination(vaccine) {
        console.log(`Providing vaccine --> ${vaccine} <-- to animal ${this._name}`);
        this._vaccines.push(vaccine);
    }
}

